blog/_posts/2021-07-31-Mitchell-Johnson_Jamie-Schmidt.md

```diff
@@ -0,0 +1,118 @@
+  
+In the config file, this means all a DynamicOrg needs is a list of names of all the Genomes and Brains it will contain.  So long as all these Genomes and Brains are properly named in the confic file, this will work seamlessly. 
+  
+## Brains / author: jamie
+-how new brain functionality was implemented before we made the switch
+-how/where the functionality lives now, how its better this way
+  
+Brains are the processors of all the Genomes in an Organism.  They are only ever accessed through their process function, which takes an emp::DataMap and then processes all the relevent information.  In DynamicOrg in particular, they are all given a DataMap containing the Genomes in the DynamicOrg, and as each Brain will know which Genomes matter for its calculations, it will then access those Genomes, process them, and write its output to the DataMap.
+  
+The Brain will be configured with the names of the Genomes it works with, along with the name of its output.  This means that you can make identical Brains which still deal with different Genomes and write their outputs to different addresses in the DataMap.  
+  
+Many Brains, such as Markov Brains, are built from Genomes themselves, and this also occurs in the process function.  This allows for Brains that evolve over time, and the Evaluator can either receive the Brain itself or whatever decision-making process it has built from the Genomes the Brain was given.
+  
+## Evaluators / author: jamie
+-how new evaluator functionality was implemented before we made the switch
+-how/where the functionality lives now, how its better this way
+  
+An Evaluator is, perhaps, the thing most users will spend the most time with for their custom experiments.  In any evolutionary situation, Organisms will need to be evaluated on whatever traits they are supposed to be evolving, and the Evaluator assigns their fitness value from whatever outputs the Brains provide.  
+  
+When coding an Evaluator, the most important thing is to make sure that, for whatever inputs your evaluate function requires, you've got Brains that can supply it.  They're configured in the same way that Brains and Genomes are, by being given a list of input names that correspond to the types that the Evaluator is expecting.
+  
+We did not make many changes to the Evaluator model, aside from making it compatible with DynamicOrgs, which in turn made them significantly easier to customize since every DynamicOrg Evaluator follows the same steps: 
+  1. Get the emp::DataMap
+  2. Get all the input values you want from it
+  3. Calculate fitness from them
+  4. Return fitness
```

Comment:
Each one of these sections is rich enough to be a post on its own! In terms of organizational flow, I think you’ll have to be careful that readers don’t lose their sense of the overall flow of the blog in the middle as they’re working through all this great detail. One solution might be to put a kind of table of contents with links into each section’s anchor headings and descriptions of what’s covered in each section (“tell them what you’re going to tell them” kind of situation). Or you could write a three or four sentence overview for each section in the main body blog post and put the full complete details on the end of the post as Appendix sections (just regular sections labeled appendix) that you could link to from the body.